lib/util/fs/index.ts

```diff
@@ -300,3 +300,10 @@ export async function getLocalFiles(
 
   return fileContentMap;
 }
+
+const validFilePathRegex = new RegExp(
+  /^(?!\/)(\.{1})?([a-z0-9/]+)?(\/\.{3})?$/
+);
+export function isValidLocalPath(s: string): boolean {
+  return validFilePathRegex.test(s);
```

Comment:
we should call `ensureValidLocalPath` to make function more save.

https://github.com/renovatebot/renovate/blob/cc33d34ca827f2acc5a00d69f38328b26ab06939/lib/util/fs/util.ts#L25

Reply:
why this is resolved without comment or code changes?

Reply:
https://github.com/jorgemmsilva/renovate/blob/feat/gogetdir/lib/modules/manager/gomod/artifacts.ts#L312

Reply:
should be inside this function

Reply:
different purpose, one is for filtering, another is for transformation.

`isValidLocalPath` - returns whether a given path is valid
 `ensureLocalPath` - returns an absolute path from a local path, I would actually argue that the name of this function is misleading

 I don't understand the benefit of mixing those.

Reply:
the idea was not to call ensure path, but do the same validation here. that probably needs some refactoring of the util functions.

that is much safer that any regex testing.

Reply:
why not both? I'm having trouble understanding the change you're requesting.
Honestly this has been opened for 2 months. I'm just trying to contribute with a feature that would be useful to me and others. Please just make the change or change things as you see fit

Reply:
ok, I'll try to add the function as a separate PR, so you only need to rebase after.

Reply:
any update on this @viceice ?

Reply:
i hopefully can do it tomorrow morning

Reply:
PR is merged, so please update this PR a hopefully last time.

Reply:
just updated (rebased) the PR, the tests still pass correctly.
Is there any extra change you would expect?